ui-lib/src/components/Switch/Switch.tsx

```diff
@@ -0,0 +1,142 @@
+import React from "react";
+import styled from "styled-components";
+import Text from "@ui/components/Text";
+import Flex from "@ui/components/Layout/Flex";
+
+const Container = styled(Flex).attrs({ alignItems: "center", flexDirection: "row" })`
+  width: fit-content;
+  column-gap: 12px;
+
+  /* reversed VARIANT */
+  &[data-reversed="true"] {
+    flex-direction: row-reverse;
+  }
+`;
+
+const Input = styled.input`
+  position: absolute;
+  visibility: hidden;
+`;
+
+const Label = styled(Text).attrs({ type: "body", fontWeight: "500" })`
+  color: currentColor;
+
+  &:first-letter {
+    text-transform: uppercase;
+  }
+
+  /* CHECKED VARIANT */
+  ${Input}:checked ~ & {
+    color: ${props => props.theme.colors.palette.v2.primary.Dark};
+  }
+`;
+
+const Switcher = styled.div`
+  --ll-switch-width: 48px;
+  --ll-switch-height: 28px;
+  --ll-switch-padding: 4px;
+
+  position: relative;
+  display: inline-block;
+
+  background: ${props => props.theme.colors.palette.v2.grey.border};
+  border-radius: 16px;
+  width: var(--ll-switch-width);
+  height: var(--ll-switch-height);
+
+  transition: background 200ms;
+  cursor: pointer;
+
+  &:before,
+  &:after {
+    content: "";
+  }
+
+  &:focus {
+    outline-style: auto;
+    outline: 1px solid ${props => props.theme.colors.palette.v2.grey.border};
+    outline-offset: 2px;
+  }
+
+  /* CIRCLE ELEMENT */
+  &:before {
+    position: absolute;
+    display: block;
+    background: ${props => props.theme.colors.palette.v2.background.default};
+    border-radius: 40px;
+
+    width: calc(calc(var(--ll-switch-width) / 2) - var(--ll-switch-padding));
+    height: calc(calc(var(--ll-switch-width) / 2) - var(--ll-switch-padding));
+    top: var(--ll-switch-padding);
+    left: var(--ll-switch-padding);
+    transition: left 0.25s;
```

Comment:
I think that using `transform: translateX` would be more performant than using `left` since it would be hardware accelerated (there are quite a lot of articles on the subject - I can link some if you are interested 🤓 ).

Reply:
Good call mate, already have some articles that talk about that in my [Pocket](https://getpocket.com). I suppose that is time to read them again 😅